query/query.go

```diff
@@ -30,12 +30,52 @@ var _ = log.Println
 type Q interface {
 	String() string
 }
-type Visibility struct {
-	Value string
+
+const (
+	OnlyPublic   uint8 = 1
+	OnlyPrivate        = 2
+	OnlyForks          = 1 << 2
+	NoForks            = 2 << 2
+	OnlyArchived       = 1 << 4
+	NoArchived         = 2 << 4
```

Comment:
The flags probably need a common prefix so we know they are related. For example `regexp/syntax` prefixes all constants in its bit flag with `Op`. I think that will lead to more understandable namespacing when using auto-complete/godoc/etc.

Reply:
I picked 'Rc'. I saw that the 'os' package uses 'O_<all caps>' for the flags, but I think 'Rc' is more conventional.